### PR TITLE
doc: Add documentation how to update template module code

### PR DESCRIPTION
**Pull request by akocbek**
### Description

Please include a summary of the change and which issue is fixed. Also, please include the motivation and context for the change. List any dependencies that are required for this change.

**Check the relevant boxes:**
- [ ] Bug fix (nonbreaking change that fixes an issue)
- [ ] New feature (nonbreaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI related update (pipeline, etc.)

### Checklist

- [ ] If relevant, a test for the change has been added or updated as part of this PR.
- [ ] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

### CODE REVIEW

File: module-assets/ci/module-template-automation/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## Updating template module content
+Most of the code for `template module` is stored in [common-dev-assets](https://github.com/terraform-ibm-modules/common-dev-assets). Therefore, to make any code changes to `template module`, files inside this root must be updated.
```
Comment (SirSpidey) [inline module-assets/ci/module-template-automation/README.md:2]:
```suggestion
# Updating the module template readme file content
Most of the code for the readme file in the `terraform-ibm-module-template` repo is stored in [common-dev-assets](https://github.com/terraform-ibm-modules/common-dev-assets). To make changes to the readme file, update the files in this directory.
```

File: module-assets/ci/module-template-automation/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+If you would like to modify any markdown files then the code changes should be added to:
+- [Modifying main README file](.terraform-docs-config-template-module.yaml)
+- [Modifying contributing section of main README file](.terraform-docs-config-template-module-contribution.yaml)
+- [Modifying tests README file](.terraform-docs-config-template-module-tests.yaml)
```
Comment (SirSpidey) [inline module-assets/ci/module-template-automation/README.md:7]:
```suggestion
To modify the Markdown content, change the following files:
- To modify most content in the main readme file, update [.terraform-docs-config-template-module.yaml](.terraform-docs-config-template-module.yaml).

    To modify the contributing section of of the main readme file, update [.terraform-docs-config-template-module-contribution.yaml](.terraform-docs-config-template-module-contribution.yaml).
- To modify the readme file in the `tests` directory, update [.terraform-docs-config-template-module-tests.yaml](.terraform-docs-config-template-module-tests.yaml).
```

File: module-assets/ci/module-template-automation/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+To update `examples` the code inside [examples](examples) should be changed.
+
+To update `tests` the code inside [tests](tests) should be changed.
```
Comment (SirSpidey) [inline module-assets/ci/module-template-automation/README.md:11]:
```suggestion
To update the default examples for the module template, update the code in the [examples](examples) directory in this rep.

To update the default tests, update the files in the [tests](tests) directory in this repo.
```

File: module-assets/ci/module-template-automation/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+[build_module_template](../../.pre-commit-config.yaml) pre-commit hook is used to build up the content for this module. The hook executes the following actions:
+- copies code, examples and tests from `common-dev-assets` into `template module` module
+- creates markdown (README) files
```
Comment (SirSpidey) [inline module-assets/ci/module-template-automation/README.md:15]:
```suggestion
The [build_module_template](../../.pre-commit-config.yaml) pre-commit hook assembles the content for the module template repo. The hook runs the following actions:
- Copies the code, examples, and tests from this `common-dev-assets` repo to the `terraform-ibm-module-template` repo.
- Creates the readme files in the `terraform-ibm-module-template` repo.
```

**Review by SirSpidey — CHANGES_REQUESTED** [review]
Stylistic changes. I don't think I changed the accuracy, but make sure.

**Comment by akocbek** [comment]
@SirSpidey I addressed your comments. Since this doc concerns both private and public module template repos. Therefore I used: `module-template` or `terraform-ibm-module-template`
an example
```
Most of the code for the readme file in the `module-template` or `terraform-ibm-module-template` repo is stored in [common-dev-assets](https://github.com/terraform-ibm-modules/common-dev-assets)....
```

**Review by SirSpidey — APPROVED** [review]
👍 Thanks for the updates.

**Comment by SirSpidey** [comment]
@ocofaigh All set to merge

**Pull request merged**